src/Schema/Column/BigIntColumnSchema.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Schema\Column;
+
+use Yiisoft\Db\Expression\ExpressionInterface;
+use Yiisoft\Db\Schema\SchemaInterface;
+
+use const PHP_INT_MAX;
+use const PHP_INT_MIN;
+
+class BigIntColumnSchema extends AbstractColumnSchema
+{
+    public function __construct(
+        string $type = SchemaInterface::TYPE_BIGINT,
+        string|null $phpType = SchemaInterface::PHP_TYPE_STRING,
+    ) {
+        parent::__construct($type, $phpType);
+    }
+
+    public function dbTypecast(mixed $value): int|string|ExpressionInterface|null
+    {
+        return match (gettype($value)) {
+            'string' => $value === '' ? null : (
+                $value <= PHP_INT_MAX && $value >= PHP_INT_MIN
+                    ? (int) $value
+                    : $value
+            ),
+            'NULL' => null,
+            'integer' => $value,
+            'boolean' => $value ? 1 : 0,
+            default => $value instanceof ExpressionInterface ? $value : (
+                ($val = (string) $value) <= PHP_INT_MAX && $val >= PHP_INT_MIN
```

Comment:
It's duplicate code with string case, may be extract it to separate method?

Reply:
The difference is to cast the value to string before check. And calling an additional method will slow down performance. Not so critical, but also not particularly necessary.